Tests/RxSwift/fromAsyncTests.swift

```diff
@@ -36,4 +44,70 @@ class FromAsyncTests: XCTestCase {
 
         XCTAssertEqual(observer.events, correct)
     }
+    
+    func testSingleResultEqualitySuccessCase() {
+        // given
+        let result = "result"
+        let expectedEvents: [Recorded<Event<String>>] = [.next(0, result), .completed(0)]
+        // when
+        _ = Single<String>
+            .fromAsync(serviceWithError)(result)
+            .asObservable()
+            .subscribe(observer)
+        scheduler.start()
+        // then
+        XCTAssertEqual(observer.events, expectedEvents)
+    }
+    
+    func testSingleOptionalResultEqualitySuccessCase() {
+        // given
+        let result: String? = "result"
```

Comment:
I would test this against a `nil` value instead of string, since it would be the exceptional case. Otherwise this looks good to me.

Reply:
Changed